### ISSUE TITLE
Maybe false positive UnnecessaryAbstractClass ?

### ISSUE THREAD

**Issue by scottkennedy** [detekt/detekt#4724]
```
data class SomeData(
    val a: String,
    val b: Int,
)

abstract class AbstractFoo {
    companion object {
        fun createSomeData(foo: Pair<String, Int>) = SomeData(foo.first, foo.second)
    }
}

class RealFoo(someData: SomeData): AbstractFoo() {
    companion object {
        fun fromSomeOtherObject(other: Any): RealFoo {
            val someData = createSomeData(Pair(other.toString(), other.hashCode()))
            return RealFoo(someData)
        }
    }
}
```

In my real case, there are many subclasses of `AbstractFoo`, and being able to call `createSomeData()` without an import is really useful.

If I try this with an interface, this doesn't work for obvious reasons (it's not a static method)
```
interface InterfaceBar {
    fun createSomeData(foo: Pair<String, Int>) = SomeData(foo.first, foo.second)
}

class RealBar(someData: SomeData): InterfaceBar {
    companion object {
        fun fromSomeOtherObject(other: Any): RealBar {
            val someData = createSomeData(Pair(other.toString(), other.hashCode()))
            return RealBar(someData)
        }
    }
}
```

## Expected Behavior
I think this is a very reasonable pattern that shouldn't trigger, but I'm open to alternatives if there's a better way of doing it.

## Observed Behavior
`An abstract class without a concrete member can be refactored to an interface. [UnnecessaryAbstractClass]`

## Steps to Reproduce
See above code

## Context
I have lots of classes that all need a data class to function.

## Your Environment
* Version of detekt used: 1.20.0
* Version of Gradle used (if applicable): 7.4.2
* Operating System and version: Various

**Comment by mhernand40** [comment]
Have you tried defining your `interface` like so?
```Kotlin
interface InterfaceBar {
    companion object {
        fun createSomeData(foo: Pair<String, Int>) = SomeData(foo.first, foo.second)
    }
}
```

I feel like that should work.

**Comment by scottkennedy** [comment]
> Have you tried defining your `interface` like so?
> 
> ```kotlin
> interface InterfaceBar {
>     companion object {
>         fun createSomeData(foo: Pair<String, Int>) = SomeData(foo.first, foo.second)
>     }
> }
> ```
> 
> I feel like that should work.

I had not tried that, but that still requires an import for `createSomeData()`.

**Comment by BraisGabin** [comment]
I agree with @mhernand40 that an interface seems the correct answer here. Other option is `open` instead of `abstract` (but this has side effects). Otherwise I recommend you to add a `@Suppress`. Or even disable the rule if you don't agree with it.

But I don't feel that this is a false positive. This rule doesn't optimice the number of imports. It finds classes that could be interfaces. And this one can be an interface.

**Comment by BraisGabin** [comment]
Closing for the above reasons. Please, don't hesitate to answer if you think that we are missing something.
